src/tools/r2rdump/R2RMethod.cs

```diff
@@ -154,51 +205,52 @@ public R2RMethod(byte[] image, MetadataReader mdReader, NativeArray methodEntryP
-                BlobReader signatureReader = mdReader.GetBlobReader(methodDef.Signature);
-                SignatureHeader header = signatureReader.ReadSignatureHeader();
-                Name = mdReader.GetString(methodDef.Name);
-                int argCount = signatureReader.ReadCompressedInteger();
-                ReturnType = new SignatureType(ref signatureReader, ref mdReader);
-                ArgTypes = new SignatureType[argCount];
-                for (int i = 0; i < argCount; i++)
-                {
-                    ArgTypes[i] = new SignatureType(ref signatureReader, ref mdReader);
-                }
-            }
-            catch (System.BadImageFormatException)
-            {
-                R2RDump.OutputWarning("The method with rowId " + rid + " doesn't have a corresponding MethodDefHandle");
-            }
+            EntryPointRuntimeFunctionId = (int)id;
         }
 
         public override string ToString()
         {
             StringBuilder sb = new StringBuilder();
 
-            if (Name != null) {
-                sb.AppendFormat($"{ReturnType.ToString()} {Name}(");
-                for (int i = 0; i < ArgTypes.Length - 1; i++)
+            if (Name != null)
+            {
+                sb.AppendFormat($"{ReturnType.ToString()} {DeclaringType}.{Name}");
+                if (IsGeneric)
                 {
-                    sb.AppendFormat($"{ArgTypes[i].ToString()}, ");
+                    sb.Append("<");
+                    int i = 0;
+                    foreach (var value in _genericParamInstance.Values)
+                    {
+                        if (i > 0)
+                        {
+                            sb.Append(", ");
+                        }
+                        sb.AppendFormat($"{Enum.GetName(typeof(GenericElementTypes), value)}");
+                        i++;
+                    }
+                    sb.Append(">");
                 }
-                if (ArgTypes.Length > 0) {
-                    sb.AppendFormat($"{ArgTypes[ArgTypes.Length - 1].ToString()}");
+
+                sb.Append("(");
+                for (int i = 0; i < ArgTypes.Length; i++)
+                {
+                    if (i > 0)
+                    {
+                        sb.Append(", ");
+                    }
+                    sb.AppendFormat($"{ArgTypes[i].ToString()}");
                 }
                 sb.Append(")\n");
             }
 
             sb.AppendFormat($"Token: 0x{Token:X8}\n");
-            foreach (RuntimeFunction runtimeFunction in NativeCode) {
+            sb.AppendFormat($"EntryPointRuntimeFunctionId: {EntryPointRuntimeFunctionId}\n");
+
+            foreach (RuntimeFunction runtimeFunction in NativeCode)
+            {
                 sb.AppendFormat($"\nStartAddress: 0x{runtimeFunction.StartAddress:X8}\n");
-                if (runtimeFunction.Size != -1) {
+                if (runtimeFunction.Size != -1)
```

Comment:
This is a personal opinion, but it may be helpful to 1) emit the number of runtime functions for each method and 2) for each runtime function emitting a `Size` line, even if the size is unknown, perhaps saying `Size: Unavailable`.